client/service/src/builder.rs

```diff
@@ -51,7 +51,71 @@ use sysinfo::{get_current_pid, ProcessExt, System, SystemExt};
 use sc_telemetry::{telemetry, SUBSTRATE_INFO};
 use sp_transaction_pool::{TransactionPool, TransactionPoolMaintainer};
 use sp_blockchain;
-use grafana_data_source::{self, record_metrics};
+use prometheus_exporter::{create_gauge, Gauge, U64, F64, Registry, PrometheusError};
+
+struct ServiceMetrics {
+	finality_height: Gauge<U64>,
+	best_height: Gauge<U64>,
+	peers_num: Gauge<U64>,
+	tx_count: Gauge<U64>,
+	node_memory: Gauge<U64>,
+	node_cpu: Gauge<F64>,
+	node_download: Gauge<U64>,
+	node_upload: Gauge<U64>,
+	sync_target: Gauge<U64>,
+}
+
+impl ServiceMetrics {
+	fn register(registry: &Registry) -> Result<Self, PrometheusError> {
+		Ok(Self {
+			finality_height: create_gauge(
+				"finality_block_height_number",
+				"Height of the highest finalized block",
+				registry
+			)?,
+			best_height: create_gauge(
+				"best_block_height_number",
+				"Height of the highest block",
+				registry
+			)?,
+			peers_num: create_gauge(
+				"peers_count",
+				"Number of network gossip peers",
+				registry
+			)?,
+			tx_count: create_gauge(
+				"transaction_count",
+				"Number of transactions",
+				registry
+			)?,
+			node_memory: create_gauge(
+				"memory_usage",
+				"Node memory usage",
+				registry
+			)?,
+			node_cpu: create_gauge(
+				"cpu_usage",
+				"Node CPU usage",
+				registry
+			)?,
+			node_download: create_gauge(
+				"receive_byte_per_sec",
```

Comment:
Not a strong opinion: I would rename this to

```suggestion
				"received_per_sec_bytes",
```

to have the unit as a suffix.